pkg/stream/output_tracer.go

```diff
@@ -25,7 +25,7 @@ func NewOutputTracer(config cfg.Config, logger log.Logger, base Output, name str
 	tracer := tracing.NewNoopTracer()
 
 	if settings.Tracing.Enabled {
```

Comment:
`tracing.ProvideTracer` already checks if `Tracing` is enabled/disabled in the configs. Only problem there is `TracerSettings` struct (`pkg/tracing/tracer.go:21`) requires `Provider` to be set. This may lead to ambiguity in my opinion since there are 2 different structs for settings but they require different fields.

I would suggest either add `Provider` field to `pkg/stream/output_configurable.go -> BaseOutputConfigurationTracing` struct or do not check if tracing is enabled in `NewOutputTracer` at all and leave the responsibility to `ProvideTracer`.